analyzer.py

```
@@ -1,8 +1,38 @@
+            candidate = text.strip()
+
+    try:
+        return json.loads(candidate)
+    except json.JSONDecodeError as exc:
+        logger.warning("Initial JSON parse failed: %s; raw response: %s", exc, text)
+        repaired = candidate.replace("{{", "{").replace("}}", "}")
+        repaired = re.sub(r",\s*([}\]])", r"\1", repaired)
+        try:
+            return json.loads(repaired)
+        except json.JSONDecodeError:
+            logger.error("Repaired JSON still invalid. Returning empty object.")
```

Ruff (0.14.13)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)